osu.Game/Users/Drawables/ClickableAvatar.cs

```diff
@@ -1,35 +1,25 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
-using System;
 using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
 using osu.Framework.Input.Events;
-using osu.Framework.Localisation;
 using osu.Game.Graphics.Containers;
-using osu.Game.Localisation;
 using osu.Game.Online.API.Requests.Responses;
+using osuTK;
 
 namespace osu.Game.Users.Drawables
 {
-    public partial class ClickableAvatar : OsuClickableContainer
+    public partial class ClickableAvatar : OsuClickableContainer, IHasCustomTooltip<UserGridPanel>
     {
-        public override LocalisableString TooltipText
-        {
-            get
-            {
-                if (!Enabled.Value)
-                    return string.Empty;
-
-                return ShowUsernameTooltip ? (user?.Username ?? string.Empty) : ContextMenuStrings.ViewProfile;
-            }
-            set => throw new NotSupportedException();
-        }
+        public ITooltip<UserGridPanel> GetCustomTooltip() => new UserGridPanelTooltip();
 
-        /// <summary>
-        /// By default, the tooltip will show "view profile" as avatars are usually displayed next to a username.
-        /// Setting this to <c>true</c> exposes the username via tooltip for special cases where this is not true.
-        /// </summary>
-        public bool ShowUsernameTooltip { get; set; }
```

Comment:
I don't believe you can just remove this. The usage in `UpdateableAvatar` is now no longer hooked up to anything. You're probably going to want this to still work, and show no hover card when specified.

Reply:
Reverted the changes, should I give the ability to change between the text tooltip and the custom tooltip?

Reply:
Sorry, then I misunderstood you previously. I removed the text tooltip with all of its references

Reply:
From my research there was 2 type of use-case for the `ClickableAvatar`:

- Displaying the _"View profile"_
- Displaying the username

From this, my understanding is that, there should be on option to show or not to show the custom tooltip. This is what I did.

Reply:
I have a question with the custom tooltip. I might be using it incorrectly. I thought that I implemented the working tooltip toggle but I have a strange behaviour.

![image](https://github.com/ppy/osu/assets/47304430/5ed9fbbc-e5da-448a-b618-6fbfbe909dfe)
I am passing the parent to the constructor so that I can check the parent when the `PopIn()` is happening.

I have 3 test users in the TestsScene: 
![image](https://github.com/ppy/osu/assets/47304430/3742ce7e-d65c-42e6-bbfc-c5c97688fa47)

They are being created the following way:
![image](https://github.com/ppy/osu/assets/47304430/d4356500-306e-411b-ae49-45583c657625)
The first two user's tooltip's enabled state is `false` and the last one's is `true`. It seems like if the tooltip **does not** resets it state:

Example: 
1. Hover over the first or second avatar -> every avatar's tooltip's `isEnabled` is `false` (It always calls back to the first time hovered avatar)
2. Hover over the last avatar -> every avatar's tooltip's `isEnabled` is `true` (It always calls back to the first time hovered avatar)